src/Sulu/Bundle/CustomUrlBundle/Resources/public/js/components/webspace/settings/custom-url/overlay/main.js

```diff
@@ -232,9 +240,62 @@ define(['underscore', 'config', 'text!./form.html'], function(_, Config, form) {
         },
 
         /**
+         * Starts the sandbox of #target-select with the given locale.
+         *
+         * @param locale
+         */
+        initializeDataSourceComponent: function (locale) {
+            if(locale === undefined) {
+                locale = this.sandbox.sulu.getDefaultContentLocale();
+            }
+
+            var $element = $('<div/>');
+            this.$find('#target-select').append($element);
+
+            this.sandbox.start(
+                [
+                    {
+                        name: 'content-datasource@sulucontent',
+                        options: {
+                            el: $element,
+                            selected: (!!this.data.targetDocument ? this.data.targetDocument.uuid : null),
+                            webspace: this.options.webspace.key,
+                            locale: locale,
+                            instanceName: 'custom-urls',
+                            rootUrl: constants.targetRootUrl,
+                            selectedUrl: constants.targetSelectedUrl,
+                            resultKey: 'nodes',
+                            selectCallback: function(id, path, title, item) {
+                                var $value = $('#custom-url-target-value');
+                                $value.val(title);
+                                $value.data('item', item);
+                                $('#custom-url-target-button-clear').show();
+
+                                var locales = this.filterConcreteLocales(this.getData().baseDomain, item.concreteLanguages);
+                                this.sandbox.emit('husky.select.target-locale.update', locales, [locales[0]], false);
+
+                                this.target = id;
+                                this.sandbox.emit('husky.overlay.custom-urls.slide-to', 0);
+                            }.bind(this)
+                        }
+                    }
+                ]
+            );
+        },
+        /**
+         * Stops and restarts CollumNavigation restarts it with correct locale.
+         *
+         * @param locale
+         */
+        changeCollumNavigationLanugage: function(locale){
```

Comment:
What do you say? Should we call this function "changeDataSourceLanguage"? In my opinion this would be more appropriate, as you re-initializing the data-source component.